src/v/compression/tests/zstd_stream_bench.cc

```diff
@@ -35,20 +37,58 @@ static inline iobuf gen(const size_t data_size) {
 inline void compress_test(size_t data_size) {
     auto o = gen(data_size);
     compression::stream_zstd fn;
+    compression::stream_zstd::init_workspace(2_MiB);
+
     perf_tests::start_measuring_time();
     perf_tests::do_not_optimize(fn.compress(std::move(o)));
     perf_tests::stop_measuring_time();
 }
 
 inline void uncompress_test(size_t data_size) {
     compression::stream_zstd fn;
+    compression::stream_zstd::init_workspace(2_MiB);
     auto o = fn.compress(gen(data_size));
+
     perf_tests::start_measuring_time();
     perf_tests::do_not_optimize(fn.uncompress(std::move(o)));
     perf_tests::stop_measuring_time();
 }
 
+inline ss::future<> async_compress_test(size_t data_size) {
+    auto o = gen(data_size);
+    compression::async_stream_zstd fn;
+    compression::async_stream_zstd::init_workspace(2_MiB);
+
+    perf_tests::start_measuring_time();
+    perf_tests::do_not_optimize(co_await fn.compress(std::move(o)));
+    perf_tests::stop_measuring_time();
+}
+
+inline ss::future<> async_uncompress_test(size_t data_size) {
+    compression::async_stream_zstd fn;
+    compression::async_stream_zstd::init_workspace(2_MiB);
+    auto o = co_await fn.compress(gen(data_size));
+
+    perf_tests::start_measuring_time();
+    perf_tests::do_not_optimize(co_await fn.uncompress(std::move(o)));
+    perf_tests::stop_measuring_time();
+}
+
 PERF_TEST(streaming_zstd_1mb, compress) { compress_test(1 << 20); }
 PERF_TEST(streaming_zstd_1mb, uncompress) { return uncompress_test(1 << 20); }
 PERF_TEST(streaming_zstd_10mb, compress) { compress_test(10 << 20); }
 PERF_TEST(streaming_zstd_10mb, uncompress) { return uncompress_test(10 << 20); }
+
+struct async_stream_zstd {};
```

Comment:
Can you note in the commit message what performance these benchmarks achieved vs. the sync versions?  We don't have a more scientific way to record results right now, but I think it's still useful to have a datapoint in the history that tells us "async was as fast as sync on CPU X at commit Y" (or what the difference was, if they aren't as fast)

Reply:
Added the benchmarking results to the commit message